File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/signus/ReplaceKeysApplyTest.java
package org.hyperledger.indy.sdk.signus;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.wallet.WalletValueNotFoundException;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.isA;

import java.util.concurrent.ExecutionException;


public class ReplaceKeysApplyTest extends IndyIntegrationTestWithSingleWallet {

	private String did;

	@Before
	public void before() throws Exception {
		did = Signus.createAndStoreMyDid(this.wallet, "{}").get().getDid();
	}

	@Test
	public void testReplaceKeysApplyWorks() throws Exception {
		Signus.replaceKeysStart(wallet, did, "{}").get();
		Signus.replaceKeysApply(wallet, did).get();
	}

	@Test
	public void testReplaceKeysApplyWorksWithoutCallingReplaceStart() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletValueNotFoundException.class));

		Signus.replaceKeysApply(wallet, did).get();
	}

	@Test
	public void testReplaceKeysApplyWorksForNotFoundDid() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletValueNotFoundException.class));

		Signus.replaceKeysStart(wallet, did, "{}").get();
		Signus.replaceKeysApply(wallet, DID1).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/SignusTests/ReplaceKeysApplyTest.cs
using Hyperledger.Indy.SignusApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{
    [TestClass]
    public class ReplaceKeysApplyTest : IndyIntegrationTestWithSingleWallet
    {
        private string _did;

        [TestInitialize]
        public async Task CreateWalletWithDid()
        {
            var result = await Signus.CreateAndStoreMyDidAsync(wallet, "{}");
            _did = result.Did;
        }        

        [TestMethod]
        public async Task TestReplaceKeysApplyWorks()
        {
            await Signus.ReplaceKeysStartAsync(wallet, _did, "{}");
            await Signus.ReplaceKeysApplyAsync(wallet, _did);
        }

        [TestMethod]
        public async Task TestReplaceKeysApplyWorksWithoutCallingReplaceStart()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletValueNotFoundException>(() =>
                Signus.ReplaceKeysApplyAsync(wallet, _did)
            );
        }

        [TestMethod]
        public async Task TestReplaceKeysApplyWorksForNotFoundDid()
        {
            await Signus.ReplaceKeysStartAsync(wallet, _did, "{}");

            var ex = await Assert.ThrowsExceptionAsync<WalletValueNotFoundException>(() =>
                Signus.ReplaceKeysApplyAsync(wallet, DID1)
            );
        }
    }
}
